vm.go

```diff
@@ -0,0 +1,131 @@
+	DirtyBytes                int64 `fileName:"dirty_bytes"`
+	DirtyExpireCentisecs      int64 `fileName:"dirty_expire_centisecs"`
+	DirtyRatio                int64 `fileName:"dirty_ratio"`
+	DirtytimeExpireSeconds    int64 `fileName:"dirtytime_expire_seconds"`
+	DirtyWritebackCentisecs   int64 `fileName:"dirty_writeback_centisecs"`
+	DropCaches                int64 `fileName:"drop_caches"`
+	ExtfragThreshold          int64 `fileName:"extfrag_threshold"`
+	HugetlbShmGroup           int64 `fileName:"hugetlb_shm_group"`
+	LaptopMode                int64 `fileName:"laptop_mode"`
+	LegacyVaLayout            int64 `fileName:"legacy_va_layout"`
+	// LowmemReserveRatio        [?]int64 `fileName:"lowmem_reserve_ratio"`
+	MaxMapCount            int64  `fileName:"max_map_count"`
+	MemoryFailureEarlyKill int64  `fileName:"memory_failure_early_kill"`
+	MemoryFailureRecovery  int64  `fileName:"memory_failure_recovery"`
+	MinFreeKbytes          int64  `fileName:"min_free_kbytes"`
+	MinSlabRatio           int64  `fileName:"min_slab_ratio"`
+	MinUnmappedRatio       int64  `fileName:"min_unmapped_ratio"`
+	MmapMinAddr            int64  `fileName:"mmap_min_addr"`
+	NrHugepages            int64  `fileName:"nr_hugepages"`
+	NrHugepagesMempolicy   int64  `fileName:"nr_hugepages_mempolicy"`
+	NrOvercommitHugepages  int64  `fileName:"nr_overcommit_hugepages"`
+	NumaStat               int64  `fileName:"numa_stat"`
+	NumaZonelistOrder      string `fileName:"numa_zonelist_order"`
+	OomDumpTasks           int64  `fileName:"oom_dump_tasks"`
+	OomKillAllocatingTask  int64  `fileName:"oom_kill_allocating_task"`
+	OvercommitKbytes       int64  `fileName:"overcommit_kbytes"`
+	OvercommitMemory       int64  `fileName:"overcommit_memory"`
+	OvercommitRatio        int64  `fileName:"overcommit_ratio"`
+	PageCluster            int64  `fileName:"page-cluster"`
+	PanicOnOom             int64  `fileName:"panic_on_oom"`
+	PercpuPagelistFraction int64  `fileName:"percpu_pagelist_fraction"`
+	StatInterval           int64  `fileName:"stat_interval"`
+	Swappiness             int64  `fileName:"swappiness"`
+	UserReserveKbytes      int64  `fileName:"user_reserve_kbytes"`
+	VfsCachePressure       int64  `fileName:"vfs_cache_pressure"`
+	WatermarkBoostFactor   int64  `fileName:"watermark_boost_factor"`
+	WatermarkScaleFactor   int64  `fileName:"watermark_scale_factor"`
+	ZoneReclaimMode        int64  `fileName:"zone_reclaim_mode"`
+}
+
+// Vm reads the Vm statistics from the specified `proc` filesystem.
+func (fs FS) Vm() (Vm, error) {
+	path := fs.proc.Path("sys/vm")
+	file, err := os.Stat(path)
+	if err != nil {
+		return Vm{}, err
+	}
+	if !file.Mode().IsDir() {
+		return Vm{}, fmt.Errorf("%s is not a directory", path)
+	}
+
+	return parseVm(path)
+}
+
+// parseVm scans predefined files in /proc/sys/vm
+// directory and gets their contents.
+func parseVm(devicePath string) (Vm, error) {
+	interfaceClass := Vm{}
+	interfaceElem := reflect.ValueOf(&interfaceClass).Elem()
```

Comment:
I would really prefer to not see reflection here either. The data isn't truly dynamic, and you've already had to type out all the file names above.

IMHO this really should be a switch statement.

Reply:
Ok, will rewrite this. The motivation was to follow the style used elswhere, as it was pointed out in the issue  

> similarly to some of the functions in the sysfs package

Reply:
I see, I'd like to get rid of reflect in this repository and have filed: https://github.com/prometheus/procfs/issues/177.

Reply:
Hm after looking around a bit apart from reflect the only solution to iterate over a struct seems to be something like

``` for _, d := range []interface{}{instance.field1, instance.field2, instance.field3} {...```

That doesn't look very sexy either, I just wait and see what you come up with in #177 ;)